icu_benchmarks/utils.py

```
@@ -0,0 +1,22 @@
+    try:
+        # Check if the input is in JSON-like format
+        if ":" in arg and '"' in arg:
+            # Wrap in curly braces to make it valid JSON
+            json_string = f"{{{arg}}}"
+            return json.loads(json_string)
+        else:
+            # Handle unquoted format
+            pairs = arg.split(',')
+            return {key.strip(): value.strip() for key, value in (pair.split(':', 1) for pair in pairs)}
+    except Exception as e:
+        raise argparse.ArgumentTypeError(f"Invalid dictionary format: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)